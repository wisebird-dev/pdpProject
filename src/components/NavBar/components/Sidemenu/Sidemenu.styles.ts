import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(
    ({ spacing, breakpoints, typography, palette }) => ({
        list: {
            width: spacing(30),
        },
        fullList: {
            [breakpoints.down('md')]: {
                width: 'auto',
            },
        },
        menuTitle: {
            color: palette.common.black,
            fontSize: 18,
            fontWeight: typography.fontWeightBold,
        },
        menuItem: {
            color: palette.common.black,
        },
        menuHeader: {
            backgroundColor: palette.background.default,
            padding: spacing(1, 1),
        },
    })
);

export default useStyles;
