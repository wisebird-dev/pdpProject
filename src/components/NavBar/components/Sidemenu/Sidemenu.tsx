import React, { useState } from 'react';
import SwipeableDrawer from '@material-ui/core/SwipeableDrawer';
import Button from '@material-ui/core/Button';
import List from '@material-ui/core/List';
import Divider from '@material-ui/core/Divider';
import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import LanguageIcon from '@material-ui/icons/Language';
import AccountCircleIcon from '@material-ui/icons/AccountCircle';
import ContactSupportIcon from '@material-ui/icons/ContactSupport';
import PaymentIcon from '@material-ui/icons/Payment';

import { Typography } from '@components';

import useStyles from './Sidemenu.styles';

const Sidemenu: React.FC = () => {
    const classes = useStyles();
    const [isOpen, setIsOpen] = useState(false);

    const toggleDrawer =
        (open: boolean) => (event: React.KeyboardEvent | React.MouseEvent) => {
            if (
                event &&
                event.type === 'keydown' &&
                ((event as React.KeyboardEvent).key === 'Tab' ||
                    (event as React.KeyboardEvent).key === 'Shift')
            ) {
                return;
            }

            setIsOpen(open);
        };

    const helpAndSettingsList = [
        {
            text: 'Language',
            icon: <LanguageIcon />,
        },
        {
            text: 'Account',
            icon: <AccountCircleIcon />,
        },
        {
            text: 'Payment Methods',
            icon: <PaymentIcon />,
        },
        {
            text: 'Contact Support',
            icon: <ContactSupportIcon />,
        },
    ];

    const MenuOptions = () => (
        <div
            className={[classes.list, classes.fullList].join(' ')}
            // role="presentation"
            onClick={toggleDrawer(false)}
            onKeyDown={toggleDrawer(false)}
        >
            <div className={classes.menuHeader}>
                <Typography variant="body1" color="textSecondary">
                    Hello, Sign in
                </Typography>
            </div>
            <List dense>
                <ListItem>
                    <Typography className={classes.menuTitle}>
                        Technology
                    </Typography>
                </ListItem>
                {['Smartphones', 'Laptops', 'Tablets', 'Gaming', 'Gadgets'].map(
                    text => (
                        <ListItem button key={text}>
                            <ListItemText
                                primary={text}
                                className={classes.menuItem}
                            />
                        </ListItem>
                    )
                )}
            </List>
            <Divider />
            <List dense>
                <ListItem>
                    <Typography className={classes.menuTitle}>
                        Help & Settings
                    </Typography>
                </ListItem>
                {helpAndSettingsList.map(({ text, icon }) => (
                    <ListItem button key={text}>
                        <ListItemIcon>{icon}</ListItemIcon>
                        <ListItemText
                            primary={text}
                            className={classes.menuItem}
                        />
                    </ListItem>
                ))}
            </List>
        </div>
    );

    return (
        <div>
            <>
                <Button onClick={toggleDrawer(true)}>toggle</Button>
                <SwipeableDrawer
                    anchor={'left'}
                    open={isOpen}
                    onClose={toggleDrawer(false)}
                    onOpen={toggleDrawer(true)}
                >
                    <MenuOptions />
                </SwipeableDrawer>
            </>
        </div>
    );
};

export default Sidemenu;
