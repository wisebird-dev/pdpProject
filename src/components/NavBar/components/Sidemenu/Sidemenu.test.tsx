import React from 'react';

import { renderWithProviders, fireEvent } from '@testUtils';

import Sidemenu from './Sidemenu';

describe('Sidemenu component', () => {
    it('matches snapshot', () => {
        const { container } = renderWithProviders(<Sidemenu />);
        expect(container.firstChild).toMatchSnapshot();
    });

    it('initially is hidden', () => {
        const { queryByText } = renderWithProviders(<Sidemenu />);
        const expectedText = 'Hello, Sign in';

        expect(queryByText(expectedText)).toBe(null);
    });

    it('opens the sidedrawer on toggle action', () => {
        const { getByText, queryByText } = renderWithProviders(<Sidemenu />);

        fireEvent.click(getByText('toggle'));

        expect(queryByText('Hello, Sign in')).toBeVisible();
    });
});
