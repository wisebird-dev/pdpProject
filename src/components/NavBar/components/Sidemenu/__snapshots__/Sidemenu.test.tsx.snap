// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidemenu component matches snapshot 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      toggle
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
  <div
    class="PrivateSwipeArea-root-6 PrivateSwipeArea-anchorLeft-7"
    style="width: 20px;"
  />
</div>
`;
